src/sentry/api/endpoints/organization_member/details.py

```diff
@@ -251,6 +263,11 @@ def put(
             if member.user == request.user and (result["role"] != member.role):
                 return Response({"detail": "You cannot make changes to your own role."}, status=400)
 
+            if organization_roles.get(result["role"]).is_retired and features.has(
+                "organizations:team-roles", organization
+            ):
+                return Response({"detail": "This role is no longer in use."}, status=400)
```

Comment:
nit: maybe f"role {result['role']} has been deprecated and is no longer in use" ?

I think including the role name in the response might be a good idea 🤷

Reply:
👍 Replaced with a more descriptive message.

Part of the reason I avoided using the role name in the first place is that the other messages avoided dropping values in, perhaps intentionally. (For example, "You do not have permission to assign the given role.") In other places this is done for ease of localization. I don't think that applies to API messages, but it may be so that API client programs can do direct equality checks against the `"detail"` value. Does anyone know about this?